substrate-node/pallets/pallet-tfgrid/src/lib.rs

```diff
@@ -1822,11 +1823,21 @@ pub mod pallet {
         ) -> DispatchResultWithPostInfo {
             T::RestrictedOrigin::ensure_origin(origin)?;
 
-            let mut farm = Farms::<T>::get(farm_id).ok_or(Error::<T>::FarmNotExists)?;
-            farm.farming_policy_limits = limits.clone();
-            Farms::<T>::insert(farm_id, farm);
+            if let Some(policy_limits) = limits {
+                let farming_policy = FarmingPoliciesMap::<T>::get(policy_limits.farming_policy_id);
+                let now = system::Pallet::<T>::block_number();
+
+                if now >= farming_policy.policy_created + farming_policy.policy_end {
+                    return Err(DispatchErrorWithPostInfo::from(
+                        Error::<T>::FarmingPolicyExpired,
+                    ));
+                }
 
-            Self::deposit_event(Event::FarmingPolicySet(farm_id, limits));
+                let mut farm = Farms::<T>::get(farm_id).ok_or(Error::<T>::FarmNotExists)?;
+                farm.farming_policy_limits = Some(policy_limits.clone());
+                Farms::<T>::insert(farm_id, farm);
+                Self::deposit_event(Event::FarmingPolicySet(farm_id, Some(policy_limits)));
```

Comment:
This clone can be removed, then, pass a reference to farm (`&farm`) for the insert, and finally, instead of passing `Some(policy_limits)` by value, pass the field on farm to which it was assigned (`farm.farming_policy_limits`).

This removes a clone, which saves an allocation and deallocation.

Reply:
thx